### PR TITLE
Add Jenkinsfile

### PR DESCRIPTION
**Pull request by loicgasser**
We now use instance roles for aws resources and integrate a Jenkinsfile for the tests.

### CODE REVIEW

**Comment by loicgasser** [comment]
jenkins test

**Comment by loicgasser** [comment]
jenkins test

**Comment by loicgasser** [comment]
jenkins test

**Comment by loicgasser** [comment]
jenkins test

**Comment by loicgasser** [comment]
jenkins test

**Comment by loicgasser** [comment]
jenkins test

**Comment by loicgasser** [comment]
jenkins test

**Comment by loicgasser** [comment]
jenkins test

**Comment by loicgasser** [comment]
jenkins test

**Comment by loicgasser** [comment]
@danduk82 let me know when instance roles are on dev/int/prod

**Comment by loicgasser** [comment]
https://github.com/geoadmin/jenkins-rancher/pull/15

**Comment by loicgasser** [comment]
@danduk82 can you add the roles on dev, int and prod?
@procrastinatio could you review? 
I want this in ASAP as PR build is currently broken in chsdi.

File: Jenkinsfile [hunk 1]
```diff
@@ -1,30 +1,35 @@
...
+      parallel (
+        'integration': {
+          sh '.venv/bin/nosetests chsdi/tests/integration/'
+        },
+        'functional': {
+          sh '.venv/bin/nosetests chsdi/tests/functional/'
+        }
+      )
     }
+  }
```
Comment (oterral) [inline Jenkinsfile]: No deploy on int or dev targets ?

Comment (loicgasser) [reply Jenkinsfile]: no not possible. We don't have the `deploy` script on this instance

**Comment by loicgasser** [comment]
errors 
are not related to this build

```

======================================================================
ERROR: chsdi.tests.integration.test_layers.test_all_searchable_layers(u'ch.bfe.energiestaedte-2000watt-aufdemweg',)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/chsdi/tests/integration/test_layers.py", line 153, in checkSearch
    resp = self.testapp.get(link, status=expectedStatus)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/webtest/app.py", line 327, in get
    expect_errors=expect_errors)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/webtest/app.py", line 636, in do_request
    self._check_status(status, res)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/webtest/app.py", line 671, in _check_status
    "Bad response: %s (not %s)", res_status, status)
AppError: Bad response: 200 OK (not 404)
-------------------- >> begin captured logging << --------------------
chsdi: DEBUG: route matched for url http://localhost/rest/services/all/SearchServer?features=ch.bfe.energiestaedte-2000watt-aufdemweg&bbox=600818.7808825106,197290.49919797093,601161.2808825106,197587.99919797093&type=featuresearch&searchText=dummy; route_name: 'search', path_info: u'/rest/services/all/SearchServer', pattern: '/rest/services/{map}/SearchServer', matchdict: {'map': u'all'}, predicates: ''
chsdi: DEBUG: debug_authorization of url http://localhost/rest/services/all/SearchServer?features=ch.bfe.energiestaedte-2000watt-aufdemweg&bbox=600818.7808825106,197290.49919797093,601161.2808825106,197587.99919797093&type=featuresearch&searchText=dummy (view name u'' against context <pyramid.traversal.DefaultRootFactory instance at 0x7f36fa427680>): Allowed (no authorization policy in use)
--------------------- >> end captured logging << ---------------------

======================================================================
ERROR: chsdi.tests.integration.test_layers.test_all_searchable_layers(u'ch.bfe.energiestaedte-2000watt-areale',)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/chsdi/tests/integration/test_layers.py", line 153, in checkSearch
    resp = self.testapp.get(link, status=expectedStatus)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/webtest/app.py", line 327, in get
    expect_errors=expect_errors)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/webtest/app.py", line 636, in do_request
    self._check_status(status, res)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/webtest/app.py", line 671, in _check_status
    "Bad response: %s (not %s)", res_status, status)
AppError: Bad response: 200 OK (not 404)
-------------------- >> begin captured logging << --------------------
chsdi: DEBUG: route matched for url http://localhost/rest/services/all/SearchServer?features=ch.bfe.energiestaedte-2000watt-areale&bbox=600818.7808825106,197290.49919797093,601161.2808825106,197587.99919797093&type=featuresearch&searchText=dummy; route_name: 'search', path_info: u'/rest/services/all/SearchServer', pattern: '/rest/services/{map}/SearchServer', matchdict: {'map': u'all'}, predicates: ''
chsdi: DEBUG: debug_authorization of url http://localhost/rest/services/all/SearchServer?features=ch.bfe.energiestaedte-2000watt-areale&bbox=600818.7808825106,197290.49919797093,601161.2808825106,197587.99919797093&type=featuresearch&searchText=dummy (view name u'' against context <pyramid.traversal.DefaultRootFactory instance at 0x7f36fa4bdfc8>): Allowed (no authorization policy in use)
--------------------- >> end captured logging << ---------------------

======================================================================
ERROR: chsdi.tests.integration.test_layers.test_all_searchable_layers(u'ch.bfe.energiestaedte-energieregionen',)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/chsdi/tests/integration/test_layers.py", line 153, in checkSearch
    resp = self.testapp.get(link, status=expectedStatus)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/webtest/app.py", line 327, in get
    expect_errors=expect_errors)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/webtest/app.py", line 636, in do_request
    self._check_status(status, res)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/webtest/app.py", line 671, in _check_status
    "Bad response: %s (not %s)", res_status, status)
AppError: Bad response: 200 OK (not 404)
-------------------- >> begin captured logging << --------------------
chsdi: DEBUG: route matched for url http://localhost/rest/services/all/SearchServer?features=ch.bfe.energiestaedte-energieregionen&bbox=600818.7808825106,197290.49919797093,601161.2808825106,197587.99919797093&type=featuresearch&searchText=dummy; route_name: 'search', path_info: u'/rest/services/all/SearchServer', pattern: '/rest/services/{map}/SearchServer', matchdict: {'map': u'all'}, predicates: ''
chsdi: DEBUG: debug_authorization of url http://localhost/rest/services/all/SearchServer?features=ch.bfe.energiestaedte-energieregionen&bbox=600818.7808825106,197290.49919797093,601161.2808825106,197587.99919797093&type=featuresearch&searchText=dummy (view name u'' against context <pyramid.traversal.DefaultRootFactory instance at 0x7f36fa6fc488>): Allowed (no authorization policy in use)
--------------------- >> end captured logging << ---------------------

======================================================================
ERROR: chsdi.tests.integration.test_layers.test_all_searchable_layers(u'ch.bfe.energiestaedte',)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/chsdi/tests/integration/test_layers.py", line 153, in checkSearch
    resp = self.testapp.get(link, status=expectedStatus)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/webtest/app.py", line 327, in get
    expect_errors=expect_errors)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/webtest/app.py", line 636, in do_request
    self._check_status(status, res)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/webtest/app.py", line 671, in _check_status
    "Bad response: %s (not %s)", res_status, status)
AppError: Bad response: 200 OK (not 404)
-------------------- >> begin captured logging << --------------------
chsdi: DEBUG: route matched for url http://localhost/rest/services/all/SearchServer?features=ch.bfe.energiestaedte&bbox=600818.7808825106,197290.49919797093,601161.2808825106,197587.99919797093&type=featuresearch&searchText=dummy; route_name: 'search', path_info: u'/rest/services/all/SearchServer', pattern: '/rest/services/{map}/SearchServer', matchdict: {'map': u'all'}, predicates: ''
chsdi: DEBUG: debug_authorization of url http://localhost/rest/services/all/SearchServer?features=ch.bfe.energiestaedte&bbox=600818.7808825106,197290.49919797093,601161.2808825106,197587.99919797093&type=featuresearch&searchText=dummy (view name u'' against context <pyramid.traversal.DefaultRootFactory instance at 0x7f36fa497560>): Allowed (no authorization policy in use)
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: chsdi.tests.integration.test_layers.test_all_ids_mapping(u'ch.bfe.energiestaedte-2000watt-aufdemweg', None, <class 'chsdi.models.vector.uvek.EnergiestaedteAufdemweg2000watt'>, Column('bgdi_id', Integer(), table=<energiestaedte_aufdemweg_2000watt>, primary_key=True, nullable=False))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/chsdi/tests/integration/test_layers.py", line 161, in checkPrimaryKeyColumnTypeMapping
    schema + '.' + model.__tablename__, layerId)
AssertionError: No feature was found in table bfe.energiestaedte_aufdemweg_2000watt for layer ch.bfe.energiestaedte-2000watt-aufdemweg

======================================================================
FAIL: chsdi.tests.integration.test_layers.test_all_ids_mapping(u'ch.bfe.energiestaedte-2000watt-areale', None, <class 'chsdi.models.vector.uvek.Energiestaedte2000wattAreale'>, Column('bgdi_id', Integer(), table=<energiestaedte_2000watt_areale>, primary_key=True, nullable=False))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/chsdi/tests/integration/test_layers.py", line 161, in checkPrimaryKeyColumnTypeMapping
    schema + '.' + model.__tablename__, layerId)
AssertionError: No feature was found in table bfe.energiestaedte_2000watt_areale for layer ch.bfe.energiestaedte-2000watt-areale

======================================================================
FAIL: chsdi.tests.integration.test_layers.test_all_ids_mapping(u'ch.bfe.energiestaedte-energieregionen', None, <class 'chsdi.models.vector.uvek.EnergiestaedteRegionen'>, Column('bgdi_id', Integer(), table=<energiestaedte_energieregionen>, primary_key=True, nullable=False))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/chsdi/tests/integration/test_layers.py", line 161, in checkPrimaryKeyColumnTypeMapping
    schema + '.' + model.__tablename__, layerId)
AssertionError: No feature was found in table bfe.energiestaedte_energieregionen for layer ch.bfe.energiestaedte-energieregionen

======================================================================
FAIL: chsdi.tests.integration.test_layers.test_all_ids_mapping(u'ch.bfe.energiestaedte', None, <class 'chsdi.models.vector.uvek.Energiestaedte'>, Column('bgdi_id', Integer(), table=<energiestaedte>, primary_key=True, nullable=False))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/.venv/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/jenkins-slave/workspace/3_PR-2774-2LHSNWLBSB7C3XO4UFR3/chsdi/tests/integration/test_layers.py", line 161, in checkPrimaryKeyColumnTypeMapping
    schema + '.' + model.__tablename__, layerId)
AssertionError: No feature was found in table bfe.energiestaedte for layer ch.bfe.energiestaedte
```

Wild update thanks @alcapat @AFoletti

**Pull request merged**